src/neo/SmartContract/StorageKey.cs

```diff
@@ -87,19 +74,6 @@ public override int GetHashCode()
             return Id.GetHashCode() + (int)Key.Span.Murmur32(0);
         }
 
-        void ISerializable.Serialize(BinaryWriter writer)
-        {
-            if (cache != null)
-            {
-                writer.Write(cache);
-            }
-            else
-            {
-                writer.Write(Id);
-                writer.Write(Key.Span);
-            }
-        }
-
         public byte[] ToArray()
         {
             if (cache is null)
```

Comment:
`Id` and `Key` must be readonly or cache could be out-synchronized

Reply:
They should be immutable. I will update these classes later, but not in this pr.